### PR TITLE
chore(CRT-222): move e2e tests to toolchain-e2e repo

### PR DESCRIPTION
**Pull request by MatousJobanek**
* This PR moves all e2e tests to the toolchain-e2e repo - see this PR: https://github.com/codeready-toolchain/toolchain-e2e/pull/1
* changes test makefile to execute e2e tests from toolchain-e2e repo
* adds a description to README describing the process and logic

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -39,18 +39,68 @@ There are a few more targets that you can find useful:
 * to remove & create the local test namespace, and create ClusterRole/ClusterRoleBinding and ServiceAccount inside of the namespace: `$ make reset-namespace`
 
 === End-to-End tests
+
+==== Background & pairing
+
+E2E tests are not located in this repository - all e2e tests are in https://github.com/codeready-toolchain/toolchain-e2e[toolchain-e2e] repo, however, it's still possible to run them locally from this repo - see <<Running End-to-End Tests>>.
```
Comment (xcoulon) [inline README.adoc]: "all e2e tests are in the... "

File: README.adoc [hunk 2]
```diff
@@ -39,18 +39,68 @@ There are a few more targets that you can find useful:
...
+
+When there is a change introduced in this repository that should be either covered by e2e tests or requires changes in the already existing tests, then all needed changes should go into the https://github.com/codeready-toolchain/toolchain-e2e[toolchain-e2e] repo.
+The logic that executes tests in openshift-ci automatically tries to pair PR opened for this (member-operator) repository with a branch that potentially exists in the developer's fork of the https://github.com/codeready-toolchain/toolchain-e2e[toolchain-e2e] repo. This pairing is based on a branch name.
```
Comment (xcoulon) [inline README.adoc]: "... to pair any PR opened on this repository "

Comment (xcoulon) [reply README.adoc]: "... This pairing is based on the current branch name."

File: README.adoc [hunk 3]
```diff
@@ -39,18 +39,68 @@ There are a few more targets that you can find useful:
...
+
+For example, if a developer with GH account `cooljohn` opens a PR (for member-operator repo) from a branch `fix-reconcile`, then the logic checks if there is a branch `fix-reconcile` also in the `cooljohn/toolchain-e2e` fork.
```
Comment (xcoulon) [inline README.adoc]: "For example, if a developer with GH a account `cooljohn`..."

Comment (xcoulon) [reply README.adoc]: "if there is a branch named `fix-reconcile`"

Comment (xcoulon) [reply README.adoc]: not sure if we should use `fork` or `clone` here, but I guess both are fine.

File: README.adoc [hunk 4]
```diff
@@ -39,18 +39,68 @@ There are a few more targets that you can find useful:
...
+If there is, then the logic:
```
Comment (xcoulon) [inline README.adoc]: "If there is a match, ..."

File: README.adoc [hunk 5]
```diff
@@ -39,18 +39,68 @@ There are a few more targets that you can find useful:
...
+
+1. clones latest changes from https://github.com/codeready-toolchain/toolchain-e2e[codeready-toolchain/toolchain-e2e]
```
Comment (xcoulon) [inline README.adoc]: "1. clones the latest changes..."

File: README.adoc [hunk 6]
```diff
@@ -39,18 +39,68 @@ There are a few more targets that you can find useful:
...
+2. fetches the `fix-reconcile` branch from `cooljohn/toolchain-e2e` fork
+3. merges `master` branch with the changes from `fix-reconcile` branch
```
Comment (xcoulon) [inline README.adoc]: "3. merges the `master` branch..."

File: README.adoc [hunk 7]
```diff
@@ -39,18 +39,68 @@ There are a few more targets that you can find useful:
...
+4. clones latest changes from https://github.com/codeready-toolchain/host-operator[host-operator] repo and builds & deploys the `host-operator` image out of it
```
Comment (xcoulon) [inline README.adoc]: "4. clones the latest changes..."

Comment (xcoulon) [reply README.adoc]: "... repo, then builds and deploys..."

File: README.adoc [hunk 8]
```diff
@@ -39,18 +39,68 @@ There are a few more targets that you can find useful:
...
+5. builds & deploys the `member-operator` image from the code that is in the PR
+6. runs e2e tests against both operators from the merged branch of the `toolchain-e2e` repo
```
Comment (xcoulon) [inline README.adoc]: "6. runes the e2e tests..."

File: README.adoc [hunk 9]
```diff
@@ -39,18 +39,68 @@ There are a few more targets that you can find useful:
...
+
+ NOTE: you can override the default namespace names where the end-to-end tests are going to be executed - eg.: `make test-e2e HOST_NS=my-host MEMBER_NS=my-member` file.
+
+===== What to do
+
+If you are still confused by the e2e location, execution and branch pairing, see the following cases and needed steps:
+
+* *Working locally:*
+** *Need to test your code using the latest version of e2e tests from https://github.com/codeready-toolchain/toolchain-e2e[toolchain-e2e] repo:*
+*** execute `make test-e2e`
```
Comment (xcoulon) [inline README.adoc]: why do you use a "level 3" list item here?

File: README.adoc [hunk 10]
```diff
@@ -39,18 +39,68 @@ There are a few more targets that you can find useful:
...
+** *Need to test your code using e2e tests located in `../toolchain-e2e` repo:*
+*** `make test-e2e-local`
 
- NOTE: you can override the default namepace names where the end-to-end tests are going to be executed - eg.: `make test-e2e HOST_NS=my-host MEMBER_NS=my-member` file.
+* *Creating a PR:*
+** *Your PR doesn't need any changes in https://github.com/codeready-toolchain/toolchain-e2e[toolchain-e2e] repo:*
```
Comment (xcoulon) [inline README.adoc]: I'm not sure why you mix underordered (`*`, `**`, etc.) and ordered (`1.`, `2.` etc) prefixes here

Comment (MatousJobanek) [reply README.adoc]: it looked better from my point of view - see the host readme https://github.com/codeready-toolchain/host-operator#what-to-do
@xcoulon Feel free to change it anyhow you want and push it to my branch ;-)

File: README.adoc (hunk 5 above)
Comment (alexeykazakov) [inline README.adoc]:
```suggestion
1. clones the latest changes from https://github.com/codeready-toolchain/toolchain-e2e[codeready-toolchain/toolchain-e2e]
```

File: README.adoc [hunk 11]
```diff
@@ -39,18 +39,68 @@ There are a few more targets that you can find useful:
...
 If you are running this tests locally on minishift, make sure that you have exposed minishift's docker-env, so that deployment can use locally built image. You can expose it by running following command.
 `eval $(minishift docker-env)`
 
 NOTE: This is not required for openshift-ci environment
 
 ==== Running End-to-End Tests
-The `make test-e2e` will take care of creating a namespace/project with a random name (or see below for enforcing the namespace). It will also create all required CRDs, role and role binding for the service account, build the Docker image for the operator and push it to the OpenShift container registry. Finally, it will deploy the operator and run the tests using
-the operator-sdk.
 
+Although the e2e tests are in the separated repository, it's still possible to run them from this repo (member-operator) and also against the current code that is at HEAD.
```
Comment (sbryzak) [inline README.adoc]: Is this referring to the current code that is at HEAD of the e2e test repo?

File: README.adoc (hunk 3 above)
Comment (tinakurian) [reply README.adoc]: @xcoulon i think you meant this: For example, if a developer with _a_ GH ~~a~~ account cooljohn... :-)

Comment (tinakurian) [reply README.adoc]: @xcoulon i think you meant this: For example, if a developer with a GH a account cooljohn... :-)

File: README.adoc (hunk 8 above)
Comment (tinakurian) [reply README.adoc]: runs* is correct here

**Pull request merged**